execution/executors/shim/process.go

```diff
@@ -35,8 +35,14 @@ func newProcess(ctx context.Context, o newProcessOpts) (*process, error) {
 	if err != nil {
 		return nil, err
 	}
+	defer func() {
```

Comment:
I'm actually fun of named returns for such purposes. Too easy to mess with the scope without them.